pkg/netpol/eval/ingressanalyzer/ingress_analyzer.go

```diff
@@ -0,0 +1,146 @@
+		UpsertObject(obj runtime.Object) error
+		// DeleteObject removes an object from the policy engine's view of the world
+		DeleteObject(obj runtime.Object) error
+	}
+)
+
+// NewIngressAnalyzer returns a new IngressAnalyzer with an empty initial state
+func NewIngressAnalyzer() *IngressAnalyzer {
+	return &IngressAnalyzer{
+		routesMap: make(map[string]map[string]*k8s.Route),
+	}
+}
+
+// NewIngressAnalyzerWithObjects returns a new IngressAnalyzer with relevant objects
+func NewIngressAnalyzerWithObjects(objects []scan.K8sObject) (*IngressAnalyzer, error) {
+	ia := NewIngressAnalyzer()
+	var err error
+	for _, obj := range objects {
+		if obj.Kind == scan.Route { // todo: when adding more objects (ingress) will rewrite if statement to switch
+			err = ia.UpsertObject(obj.Route)
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+	return ia, err
+}
+
+// GetIngressControllerPod creates a new ingress-controller pod, as the source of all ingress connections
+func (ia *IngressAnalyzer) GetIngressControllerPod() eval.Peer {
+	ingressPod := &k8s.Pod{
+		Name:      "ingress-controller",
+		Namespace: "",
+	}
+	return &k8s.WorkloadPeer{Pod: ingressPod}
+}
+
+// UpsertObject updates (an existing) or inserts (a new) object in the ingress analyzer
+func (ia *IngressAnalyzer) UpsertObject(rtobj runtime.Object) error {
+	obj := rtobj.(*ocroutev1.Route)
+	// route object
+	if obj != nil { // todo: when adding more objects (ingress) will rewrite if statement to switch on rtobj.(type)
+		return ia.upsertRoute(obj)
+	}
+	return nil
+}
+
+// DeleteObject removes an object from the ingress analyzer
+func (ia *IngressAnalyzer) DeleteObject(rtobj runtime.Object) error {
+	obj := rtobj.(*ocroutev1.Route)
+	// route object
+	if obj != nil { // todo: when adding more objects (ingress) will rewrite if statement to switch on rtobj.(type)
+		return ia.deleteRoute(obj)
+	}
+	return nil
+}
+
+// ClearResources: deletes all current ingress resources
+func (ia *IngressAnalyzer) ClearResources() {
+	ia.routesMap = make(map[string]map[string]*k8s.Route)
+}
+
+func (ia *IngressAnalyzer) upsertRoute(rt *ocroutev1.Route) error {
+	routeObj, err := k8s.RouteFromOCObject(rt)
+	if err != nil {
+		return err
+	}
+	if _, ok := ia.routesMap[routeObj.Namespace]; !ok {
+		ia.routesMap[routeObj.Namespace] = make(map[string]*k8s.Route)
+	}
+	ia.routesMap[routeObj.Namespace][routeObj.Name] = routeObj
+	return nil
+}
+
+func (ia *IngressAnalyzer) deleteRoute(rt *ocroutev1.Route) error {
+	if rtMap, ok := ia.routesMap[rt.Namespace]; ok {
+		delete(rtMap, rt.Name)
+		if len(rtMap) == 0 {
+			delete(ia.routesMap, rt.Namespace)
+		}
+	}
+	return nil
+}
+
+////////////////////////////////////////////////////////////////////////////////////
+
+// AllowedIngressConnectionsToAWorkloadPeer return the allowed external ingress-controller's connections to the dst peer
+func (ia *IngressAnalyzer) AllowedIngressConnectionsToAWorkloadPeer(dst eval.Peer, pe *eval.PolicyEngine) (eval.Connection, error) {
```

Comment:
- used in the `ingress_analyzer_test.go`
- to be used also in `connlist.getConnectionsList()`
- gets a `Peer` as a parameter to be compatible with the idea of `AllAllowedConnectionsBetweenWorkloadPeers` and for cases of focus-workload used